doc/Language/syntax.pod6

```diff
@@ -223,109 +223,214 @@ say 'code again';
-of statements, giving a name to objects, data structures and blocks of
-code. These names must start with an alphabetic character (or an
-underscore), followed by zero or more word characters (alphabetic,
-underscore or number). You can also embed dashes C<-> or single quotes
-C<'> in the middle, but not two in a row, and only if followed
-immediately by an alphabetic character.
+Identifiers are grammatical building blocks that may be used to give a name
+to entities/objects such as constants, variables (e.g. Scalars) and routines
+(e.g. Subs and Methods). In a variable name, an identifier is normally preceded
+by a sigil; i.e. the sigil is not part of the identifier, only of the variable
+name.
+
+    constant c = 299792458;         # Identifier "c" names an Int
+    my $a = 123;                    # Identifier "a" in the name "$a" of a Scalar
+    sub hello { say "Hello!" };     # Identifier "hello" names a Sub
+
+Identifiers come in different forms: ordinary identifiers, extended
+identifiers, and compound identifiers.
+
+=head3 Ordinary identifiers
+
+An ordinary identifier is composed of a leading alphabetic character
+that may be followed by one or more alphanumeric characters. It may also
+contain isolated, embedded apostrophes C<'> and/or hyphens C<->, provided
+that the next character is each time alphabetic.
+
+The definitions of "alphabetic" and "alphanumeric" include appropriate Unicode
+characters. Which characters are "appropriate" depends on the implementation.
+In the Rakudo/MoarVM Perl 6 implementation alphabetic characters include
+Unicode characters in the character category I<Letter>, and the underscore
+C<_>. Alphanumeric characters additionally include Unicode characters in the
+general category I<Number, Decimal Digit> (Nd).
 
 =begin code :skip-test
-# valid identifiers:
+# valid ordinary identifiers:
 x
 something-longer
 with-numbers1234
-don't
-fish-food
-π
+don't-do-that
+piece_of_π
+駱駝道                 # "Rakuda-dō", Japanese for "Way of the camel"
 =end code
 
 =begin code :skip-test
-# not valid identifiers:
-with-numbers1234-5
-42
-is-prime?
-fish--food
+# invalid ordinary identifiers:
+42                    # Identifier does not start with alphabetic character
+with-numbers1234-5    # Embedded hyphen not followed by alphabetic character
+is-prime?             # Question mark is not alphanumeric
+x²                    # Superscript 2 has Unicode general category No
 =end code
 
-You use identifiers to name of constants, types (including classes and
-modules) and routines (subs and methods); they also appear in variable
-names usually proceeded by a sigil; see L<variables|/language/variables>
-for more details.
+In the C<x²> example above, since the Unicode superscript numeral C<²> is not
+alphanumeric, it cannot be a part of an ordinary identifier. Still, it may form
+part of a valid expression as an operator. The expression C<$x²>, for instance,
```

Comment:
As long as it's clear it's not alphanumeric, the rest of the paragraph is probably not needed in a section about identifiers, not even as a teaser for the other section

Reply:
And since it is the first comment, I would really have appreciated if you had not closed the other PR, added new commits addressing (or not) my comments, which you apparently have to a extent. Please do so this time; just push new commits, and comment responding to these comments.